libnetwork/etchosts/fuzz_test.go

```diff
@@ -0,0 +1,38 @@
+package etchosts
+
+import (
+	"os"
+	"testing"
+
+	fuzz "github.com/AdaLogics/go-fuzz-headers"
+)
+
+func FuzzAdd(f *testing.F) {
+	f.Fuzz(func(t *testing.T, data []byte) {
+		ff := fuzz.NewConsumer(data)
+		fileBytes, err := ff.GetBytes()
+		if err != nil {
+			return
+		}
+		noOfRecords, err := ff.GetInt()
+		if err != nil {
+			return
+		}
+
+		recs := make([]Record, 0)
+		for i := 0; i < noOfRecords%40; i++ {
+			r := Record{}
+			err = ff.GenerateStruct(&r)
+			if err != nil {
+				return
+			}
+			recs = append(recs, r)
+		}
+		tmpDir := t.TempDir()
+		err = os.WriteFile(filepath.Join(tmpDir, "testFile"), fileBytes, 0644)
+		if err != nil {
+			return
+		}
+		Add("testFile", recs)
```

Comment:
Looks like the `Add("testFile", recs)` should now also be updated; perhaps we should use a variable for the filename. While updating, perhaps also switch to the new `0o644` format;

```suggestion
		tmpDir := t.TempDir()
		testFile := filepath.Join(tmpDir, "testFile")
		err = os.WriteFile(testFile, fileBytes, 0o644)
		if err != nil {
			return
		}
		_ = Add(testFile, recs)
```